src/main/java/org/mule/extension/http/api/request/authentication/UsernamePasswordAuthentication.java

```diff
@@ -57,19 +58,23 @@ public boolean shouldRetry(Result<Object, HttpResponseAttributes> firstAttemptRe
 
   @Override
   public String getUsername() {
-    return username;
+    return username.resolve();
   }
 
   @Override
   public String getPassword() {
-    return password;
+    return password.resolve();
   }
 
   @Override
   public boolean isPreemptive() {
     return preemptive;
   }
 
+  public boolean forceConnectionClose() {
+    return username.getExpression().isPresent() || password.getExpression().isPresent();
```

Comment:
Please make double sure this works as expected when the value is static.

Reply:
Just checked. Works.